jobs/kpi-forecasting/kpi_forecasting/pandas_extras.py

```diff
@@ -26,4 +26,32 @@ def aggregate_to_period(
 
     x = df.copy(deep=True)
     x[date_col] = pd.to_datetime(x[date_col]).dt.to_period(period[0]).dt.to_timestamp()
-    return x.groupby(date_col).agg(aggregation).reset_index()
+
+    # treat numeric and string types separately
+    x_string = x.select_dtypes(include=["datetime64", object])
+    x_numeric = x.select_dtypes(include=["float", "int", "datetime64"])
+
+    if set(x_string.columns) | set(x_numeric.columns) != set(x.columns):
```

Comment:
Can you explain why this check is necessary? I can't think of an instance in either `ProphetForecast` or `FunnelForecast` that might cause this check to fail.

Reply:
It's a fail-safe.  The idea is that this function doesn't cover all types so if an expected one appears it will throw an error instead of quietly doing something weird (There whole motivation for adding tests for this function was it essentially "quietly did something weird*" when I updated the code).  IE if a new column was boolean valued this would throw an error.

* The string-valued columns got aggregated like `+` so you'd see values like `desktopdesktopdesktopdesktopdesktop`, it was funny